Tiltfile

```diff
@@ -0,0 +1,203 @@
+"""
+
+additional_docker_build_commands = """
+COPY --from=tilt-helper /usr/bin/docker /usr/bin/docker
+COPY --from=tilt-helper /go/kubernetes/client/bin/kubectl /usr/bin/kubectl
+"""
+
+##############################
+# Setup Velero
+##############################
+
+# Set up a local_resource build of the Velero binary. The binary is written to _tiltbuild/velero.
+local_resource(
+    "velero_server_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild;PKG=. BIN=velero GOOS=linux GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild ./hack/build.sh',
+    deps = ["cmd", "internal", "pkg"],
+    ignore = ["pkg/cmd"],
+)
+
+local_resource(
+    "velero_local_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild/local;PKG=. BIN=velero GOOS=' + local_goos + ' GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild/local ./hack/build.sh',
+    deps = ["internal", "pkg/cmd"],
+)
+
+tilt_dockerfile_header = """
+FROM gcr.io/distroless/base:debug as tilt
+WORKDIR /
+COPY --from=tilt-helper /start.sh .
+COPY --from=tilt-helper /restart.sh .
+COPY velero .
+"""
+
+dockerfile_contents = "\n".join([
+    tilt_helper_dockerfile_header,
+    additional_docker_helper_commands,
+    tilt_dockerfile_header,
+    additional_docker_build_commands,
+])
+
+ # Set up an image build for Velero. The live update configuration syncs the output from the local_resource
+ # build into the container.
+docker_build(
+    ref = "velero/velero",
+    context = "_tiltbuild/",
+    dockerfile_contents = dockerfile_contents,
+    target = "tilt",
+    entrypoint=["sh", "/start.sh", "/velero"],
+    live_update=[
+        sync("./_tiltbuild/velero", "/velero"),
+        run("sh /restart.sh"),
+    ])
+
+##############################
+# Setup plugins
+##############################
+
+providers = {}
+
+def load_provider_tiltfiles():
+    all_providers = settings.get("providers", []) 
+    enable_providers = settings.get("enable_providers", [])
+
+    ## Load settings only for providers to enable
+    for name in enable_providers:
+        repo = all_providers[name] 
+        file = repo + "/tilt-provider.json"
+        if not os.path.exists(file):
+            print("Provider settings not found for %s:", name)
+            print("Please ensure this plugin repository has a tilt-provider.json file included.")
+            continue
+        provider_details = read_json(file, default = {})
+        if type(provider_details) != type([]):
+            if "context" in provider_details:
+                provider_details["context"] = repo + "/" + provider_details["context"]
+            else:
+                provider_details["context"] = repo
+            if "go_main" not in provider_details:
+                provider_details["go_main"] = "main.go"
+            providers[name] = provider_details
+
+# Enable each provider
+def enable_providers():
+    if not providers:
+        print("No providers to enable.")
+        return
+    for name in providers:
```

Comment:
It depends on what we want to do. I'd like it for the developer to be able to choose to load from 0.* providers as a Tilt resource. Right now this is done in the `enable_providers` list in the tilt-settings.json. I'm not seeing how this would be possible with this suggestion. Please clarify this part if I missed it.

Reply:
Sorry, I don't think I did great job explaining what I meant. Let me try again! :)

At this point when we are iterating over `providers`, it is a dict with `name` as the key and will only contain providers that the user has enabled in their Tilt settings since we filtered them out in `load_provider_tiltfiles`, e.g.
```
{
  "aws": {
    "plugin_name": "velero-plugin-for-aws",
    ...
  }
}
```

Instead, I am proposing that we modify `load_provider_tiltfiles` so that this `providers` variable is a list. e.g. in `load_provider_tiltfiles` we could have:
```
providers = []

for name in enable_providers:
  ...
  providers.append(provider_details)
```

That would mean that the `providers` variable would now look like: `[{"plugin_name": "velero-plugin-for-aws", ...}]`. Then when we iterate over it, we could pass that whole object into `enable_providers`:
```
for provider in providers:
  enable_provider(provider)
```
Then in `enable_provider` on line 154, we don't need to call `providers.get(name)` as the parameter to the function _is_ the provider. Does that make sense? As I mentioned in my original comment though, this is only a suggestion just to try and limit the scope where the global variable `providers` needed to be accessed.

Reply:
Thanks for the expanded explanation, I needed it obviously. It's a better approach, changed it.

Reply:
Thanks for making these changes! I've been stung by global variables getting modified in ways I don't expect and it being a real pain to figure out what's going on :weary: